common/src/main/java/com/wynntils/features/tooltips/ItemCompareFeature.java

```diff
@@ -4,162 +4,590 @@
+        tooltips.forEach(tooltip -> tooltip.offsetX(offsetX));
+        hoveredX += offsetX;
+
+        event.setCanceled(true);
+
+        changePositioner = true;
+        poseStack.pushPose();
+        poseStack.scale(hoveredScaleFactor, hoveredScaleFactor, 1);
+        guiGraphics.renderTooltip(
+                font, hoveredLines, hoveredItemStack.getTooltipImage(), (int) (hoveredX / hoveredScaleFactor), (int)
+                        (hoveredY / hoveredScaleFactor));
         poseStack.popPose();
-    }
 
-    private boolean isMatchingType(ItemStack itemStack, GearTypeItemProperty gearItemReference) {
-        Optional<GearTypeItemProperty> gearOpt = Models.Item.asWynnItemProperty(itemStack, GearTypeItemProperty.class);
-        if (gearOpt.isEmpty()) return false;
+        for (Tooltip tooltip : tooltips) {
+            poseStack.pushPose();
+            float scaleFactor = tooltip.getScaleFactor();
+            poseStack.scale(scaleFactor, scaleFactor, 1);
+            guiGraphics.renderTooltip(
+                    font,
+                    tooltip.getLines(),
+                    tooltip.getVisualTooltipComponent(),
+                    (int) (tooltip.getX() / scaleFactor),
+                    (int) (tooltip.getY() / scaleFactor));
+            poseStack.popPose();
+        }
+        changePositioner = false;
 
-        return gearOpt.get().getGearType() == gearItemReference.getGearType();
+        poseStack.popPose();
     }
 
-    @SubscribeEvent
-    public void onScreenClose(ScreenClosedEvent event) {
-        compareToEquipped = false;
+    private boolean isMatchingType(WynnItem wynnItem, GearTypeItemProperty gearItemReference) {
+        if (wynnItem instanceof GearItem gearItem) {
+            return gearItem.getGearType() == gearItemReference.getGearType();
+        }
+        return false;
     }
 
-    private void onCompareModeKeyPress(Slot hoveredSlot) {
-        if (comparedItem != null) {
-            comparedItem = null;
-            compareToEquipped = true;
-        } else {
-            compareToEquipped = !compareToEquipped;
-        }
+    private boolean isMatchingType(ItemStack itemStack, GearTypeItemProperty gearItemReference) {
+        Optional<GearTypeItemProperty> gearOpt = Models.Item.asWynnItemProperty(itemStack, GearTypeItemProperty.class);
+        return gearOpt.isEmpty() ? false : gearOpt.get().getGearType() == gearItemReference.getGearType();
     }
 
     private void onSelectKeyPress(Slot hoveredSlot) {
         if (hoveredSlot == null) return;
+        // Selecting items behaves weirdly on tm updates, so no selecting in tm
+        if (Models.Container.getCurrentContainer() instanceof TradeMarketContainer) return;
+        int slot = hoveredSlot.getContainerSlot();
+
+        // Selecting armor and accessory slots is not allowed
+        if (slot == 9 || slot == 10 || slot == 11 || slot == 12 || slot == 36 || slot == 37 || slot == 38 || slot == 39)
```

Comment:
And long ifs should also have brackets.

Reply:
Also, I wonder, shouldn't InventoryModel keep track of these slots? I don't think it's great that they are hard-coded here, in such an "easy-to-miss" way.